nengo/spa/spa.py

```diff
@@ -76,12 +78,20 @@ def __init__(self):
                 self.thal = spa.Thalamus(self.bg)
     """
 
-    def __init__(self, label=None, seed=None, add_to_container=None):
+    def __init__(self, label=None, seed=None, add_to_container=None,
+                 vocabs=[]):
         super(SPA, self).__init__(label, seed, add_to_container)
         enable_spa_params(self)
         self._modules = {}
         self._default_vocabs = {}
 
+        for vo in vocabs:
+            if vo.dimensions in self._default_vocabs:
+                warnings.warn("Duplicate vocabularies with dimension:%i. "
```

Comment:
I had no idea `%i` is a thing... we usually use `%d` for integers. What's the difference between the two?

Also I'd also lose the colon there, we don't use it in any other place that we print dimensionality.